GVRf/Framework/framework/src/main/java/org/gearvrf/GVRPicker.java

```diff
@@ -21,29 +21,256 @@
+        mScene = scene;
+        mHasChanged = true;
+        mType = TYPE_PICKMANAGER;
+    }
+
+    static public long getComponentType()
+    {
+        return TYPE_PICKMANAGER;
     }
 
     /**
-     * Casts a ray into the scene graph, and returns the objects it intersects.
+     * Get the current ray to use for picking.
+     * 
+     * If the picker is attached to a scene object,
+     * this ray is derived from the scene object's transform.
+     * The origin of the ray is the translation component
+     * of the total model matrix and the ray direction
+     * is the forward look vector.
+     * 
+     * @return pick ray
+     */
+    public float[] getPickRay()
+    {
+        mPickRay[0] = mRayOrigin.x;
+        mPickRay[1] = mRayOrigin.y;
+        mPickRay[2] = mRayOrigin.z;
+        mPickRay[3] = mRayDirection.x;
+        mPickRay[4] = mRayDirection.y;
+        mPickRay[5] = mRayDirection.z;
+        return mPickRay;
+    }
+    
+    /**
+     * Gets the current pick list.
+     * 
+     * Each collision with an object is described as a
+     * GVRPickedObject which contains the scene object
+     * and collider hit, the distance from the camera
+     * and the hit position in the coordinate system
+     * of the collision geometry. The objects in the pick
+     * list are sorted based on increasing distance
+     * from the origin of the pick ray.
+     * @return GVRPickedObject array with objects picked or null if nothing picked.
+     * @see doPick
+     * @see IPickEvents
+     * @see setPickRay
+     */
+    public final GVRPickedObject[] getPicked()
+    {
+        return mPicked;
+    }
+    /*
+     * Sets the origin and direction of the pick ray.
+     * 
+     * If the picker is not attached to a scene object but you
+     * still want to get pick events, you must set the pick
+     * ray manually with this function and call {@link doPick}.
+     * 
+     * @param ox    X coordinate of origin in world coordinates.
+     * @param oy    Y coordinate of origin in world coordinates.
+     * @param oz    Z coordinate of origin in world coordinates.
+     * @param dx    X coordinate of ray direction in world coordinates.
+     * @param dy    Y coordinate of ray direction in world coordinates.
+     * @param dz    Z coordinate of ray direction in world coordinates.
+     * @see doPick
+     */
+    public void setPickRay(float ox, float oy, float oz, float dx, float dy, float dz)
+    {
+        mRayOrigin.x = ox;
+        mRayOrigin.y = oy;
+        mRayOrigin.z = oz;
+        mRayDirection.x = dx;
+        mRayDirection.y = dy;
+        mRayDirection.z = dz;
+        mHasChanged = true;
+    }
+    
+    public void onDrawFrame(float frameTime)
+    {
+        if (getOwnerObject() != null)
+        {
+            GVRTransform trans = getOwnerObject().getTransform();
+            Matrix4f worldmtx = trans.getModelMatrix4f();
+            worldmtx.getTranslation(mRayOrigin);
+            mRayDirection.x = 0;
```

Comment:
If you attach the picker to a scene object it will automatically compute the ray for you. setPickRay is for when you want to generate pick events from a particular ray but it is not derived from a scene object. For example, if you want to generate a ray based on the mouse position you can have a GVRPicker which will generate pick events even though it is not in the scene.